src/group/controller/collection.ts

```diff
@@ -14,13 +15,39 @@ class GroupMemberCollectionControlller extends Controller {
      * Checks if user type is a group
      */
     if (!groupService.isGroup(user)) {
-      throw new BadRequest('User must be a group to gain access!!');
+      throw new BadRequest('This endpoint only exists for groups');
     }
 
     const members = await groupService.findAllGroupMembers(user);
     ctx.response.body = hal.collection(user, members);
   }
 
+  async post(ctx: Context) {
+
+    const group = await userService.findById(parseInt(ctx.state.params.id, 10));
+
+    if (!groupService.isGroup(group)) {
+      throw new BadRequest('This endpoint only exists for groups');
+    }
+
+    const userBody: any = ctx.request.body;
+
+    try {
+      await groupService.findByUserId(parseInt(userBody.userId, 10));
+      throw new Conflict('User already part of this group');
+    } catch (err) {
+      if (!(err instanceof NotFound)) {
+        throw err;
+      }
+    }
+
+    await groupService.save(userBody.userId, group);
+
+    ctx.response.status = 201;
+    ctx.response.headers.set('Location', '/user/' + group.id + '/member');
+  }
```

Comment:
This suggests that a new resource has been created with this location, and that's not really true.